scripts/detect-host-ip.sh

```
@@ -0,0 +1,178 @@
+# --- Mirrored networking gate (WSL2 only) ----------------------------------
+# Checked *before* .env is touched. Without mirrored networking WSL2 sits on
+# its own NAT, so everything detected above is WSL's address rather than the
+# one guests need -- and writing it would replace a correct hand-set
+# HOST_LAN_IP with an address nobody on the LAN can reach.
+wsl_mirrored_off() {
+  is_wsl || return 1
+  # powershell.exe is only used to locate %UserProfile% -- the Windows user
+  # name need not match the WSL one, so /mnt/c/Users/$USER is not reliable.
+  local wslconfig
+  wslconfig=$(powershell.exe -NoProfile -Command \
+    'if (Test-Path "$env:UserProfile\.wslconfig") { Get-Content "$env:UserProfile\.wslconfig" -Raw }' \
```

Shellcheck (0.11.0)

[info] Expressions don't expand in single quotes, use double quotes for that.

(SC2016)